### ISSUE TITLE
Solution for knowing when a user's User Time Offset has the correct value after they join a session

### ISSUE THREAD

**Issue by Zyzyl** [Neos-Metaverse/NeosPublic#3062]
### Is your feature request related to a problem? Please describe.

I would like to display the time at which a user joined a session in that user's local time. In order to do this, I've tried saving the joining user's Local Time Offset on the impulse from the User Joined node. Unfortunately, I've found it takes a variable (potentially long, e.g. over 30-40 seconds) amount of time before the correct data is actually available from the node. A similar issue is also present if using an impulse from e.g. OnStart. I also considered trying User Time - UTC Now, but that also seems to be delayed. These delays are present even if the user is the host of the session.

### Relevant issues

Some of the relevant nodes were added as a result of https://github.com/Neos-Metaverse/NeosPublic/issues/1259

### Describe the solution you'd like

I can userstand that this information isn't necessarily high priority, so it may not be available from super early impulses like UserJoined / OnStart. However it would be helpful to me to have some method of detecting when the node has the correct information. 

- One possibility would be an event node which fires an impulse.
- Another possibility would a bool output from the User Time Offset node which outputs True if the correct Local Time Offset data is available.
- A guarantee that the User Time Offset will be correct after x seconds / updates, that way I could use a delayed impulse. 

### Describe alternatives you've considered

I've considered continually checking (at reasonable intervals) whether User Time Offset != 0 after they join, until that becomes true. However, that's fragile as some users may actually have a User Time Offset of 0

**Comment by shadowpanther** [comment]
It's odd that the time offset is not propagated to the host and other users immediately. Could it be that the user does not sent any streams/info to the host until they actually spawn? If you try to use the `User Spawn` impulse to check the time offset, is it populated by then?

**Comment by Frooxius** [comment]
Yes, user doesn't start updating until the initial sync is complete. However I'd recommend against this approach completely, because the offset can potentially change even after the user has joined if they change their clock - it's rare, but it can happen.

Instead I'd recommend to simply save the world time "T" when they join. If you subtract the current T from this saved one, you get the time offset of when they have joined relative to the current time. This will always be the same and you can use it to always recalculate the time based on the current offset.

Alternatively one thing you could try is check for when "IsPresentInWorld" becomes true for the first time for given user, as that usually indicates that the initial sync is complete and they have "activated" the world.
